Sprint three

AliRasikh opened the pull request.

maggie16t committed:
new installation after connection with backend for my (maggie) branch

Commit:
add the UUID

maggie16t committed:
navbar and some design changes

mtug812 committed:
latest frontend + maggie's branch merged

Commit:
flake8 problem is solved

Commit:
login with jwt

Commit:
+msg for login

Commit:
.

Commit:
nothing

khadijamami committed:
Merge branch 'khadija_ali' into dija_s3

khadijamami committed:
Merge pull request #43 from AliRasikh/dija_s3

Dija s3

AliRasikh committed:
jwt token added in login

AliRasikh committed:
Merge branch 'khadija_ali' of github.com:AliRasikh/data-scraping-application into s3_ali

AliRasikh committed:
Merge pull request #46 from AliRasikh/khadija_ali

Bckend Khadija and ali - jwt token auth added

mtug812 committed:
login done;have to reinstall talwind;next step-must test with the latest backend; signuppage -to be further developed by maggie

mtug812 committed:
merged latest backend from sprint_three

mtug812 committed:
login works; styling is missing + files in the backend to create dummy user

AliRasikh committed:
test version for frontend

AliRasikh committed:
test version for the frontend

mtug812 committed:
history login navbar edited

mtug812 committed:
backend edited

maggie16t committed:
sign up page and styling

AliRasikh committed:
test branch

maggie16t committed:
updatet signup page

mtug812 committed:
merged with maggie's branch(signup)

mtug812 committed:
small change

AliRasikh committed:
Sprint3

AliRasikh committed:
Merge pull request #49 from AliRasikh/s3_ali

Sprint3_ali

mtug812 committed:
functionality is working perfectly; tailwind reinstalled;style adjustment still needed

mtug812 committed:
prettier

mtug812 committed:
small changes

mtug812 committed:
latest frontend

TiboBruneel commented on backend/app.py:
```diff
@@ -1,37 +1,89 @@
+# route to test jwt auth
+@app.route("/auth")
+@token_required
+def auth():
```
Is this required?

TiboBruneel commented on backend/core/scraper.py:
```diff
@@ -64,33 +72,118 @@ def scrape_with_bs4(url: str):
+            formatted_lines.append(f"- {line}")  # Bullet points
 
+    # Convert formatted lines to a readable output
+    readable_output = "\n".join(formatted_lines)
```
Very nice work 🚀

TiboBruneel commented on frontend/src/components/ErrorHandler.tsx:
```diff
@@ -0,0 +1,13 @@
+// export const handleError = (error:any) => {
```
Still needed?

TiboBruneel commented on frontend/src/api/globalvariables.ts:
```diff
@@ -1,5 +1,2 @@
-
-
-
+export const BASE_URL = "http://127.0.0.1:5000";
```
Maybe fit for a .env variable

TiboBruneel commented on frontend/src/pages/HistoryPage.tsx:
```diff
@@ -0,0 +1,143 @@
+      setLoading(true);
+
+      const token = localStorage.getItem("authToken");
+      console.log("Token stored:", token ? token.substring(0, 15) + "..." : "No token");
```
Maybe not the wisest to log tokens

TiboBruneel commented on frontend/src/pages/HistoryPage.tsx:
```diff
@@ -0,0 +1,143 @@
+      } catch (authErr) {
+        console.error("Auth test failed:", authErr);
+        throw new Error("Token validation failed");
+      }
```
👍

TiboBruneel commented on frontend/src/pages/HistoryPage.tsx:
```diff
@@ -0,0 +1,143 @@
+    } finally {
+      setLoading(false);
+    }
+  };
```
Might also not be so good to log too much in the front-end browser

TiboBruneel commented on frontend/vite.config.ts:
```diff
@@ -1,11 +1,11 @@
   ],
-})
\ No newline at end of file
+});
```
Very nice progress on front-end this sprint, great work 🚀

mtug812 committed:
adjusted styling with tailwind

mtug812 committed:
adjusted styling with tailwind

mtug812 committed:
merged backend changes from sprint_three

mtug812 committed:
selenium implemented in FE

mtug812 committed:
history - method added

mtug812 committed:
scrape method added in the history list

mtug812 committed:
-global variable stored in ENV
- historyPage: removed some console logs
-scrapepage: errors and removed some logs
-logoutbutton: removed href

Commit:
backend dev documentation + scr.method in history

Commit:
documentation changes

Commit:
frontend dev documentation

mtug812 committed:
comment removed

mtug812 committed:
add signup from Maggie

mtug812 committed:
merge rontend changes from s3_marian

mtug812 committed:
updated scrapepage text from <<static>> to <<static and dynamic websites>>

Commit:
frontend documentation

Commit:
frontend documentation changed

Commit:
again

AliRasikh committed:
s3 update

AliRasikh committed:
update docs

AliRasikh committed:
Merge branch 'sprint_three' of github.com:AliRasikh/data-scraping-application into sprint_three

AliRasikh committed:
Merge branch 'sprint_three' of github.com:AliRasikh/data-scraping-application into sprint_three

AliRasikh committed:
update docs
